src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,111 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String request = "INSERT INTO manufacturers(name, country) VALUES(?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(request,
+                             Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert manufacturer to DB. "
+                    + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        String request = "SELECT * FROM manufacturers WHERE id = ? AND is_deleted = false";
+        try (PreparedStatement statement = getPreparedStatement(request)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            manufacturers = getManufacturers(resultSet);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer from DB. ID: " + id, e);
+        }
+        return manufacturers.size() == 0 ? Optional.empty() : Optional.of(manufacturers.get(0));
+    }

```

Comment:
Можно и так, но это не логично использовать List заведомо зная что у нас будет в результате только один объект. Давай лучше так:
```suggestion
        Manufacturer manufacturers = new Manufacturer();
        String request = "SELECT * FROM manufacturers WHERE id = ? AND is_deleted = false";
        try (PreparedStatement statement = getPreparedStatement(request)) {
            statement.setLong(1, id);
            ResultSet resultSet = statement.executeQuery();
            if (resultSet.next()){
            manufacturers = getManufacturers(resultSet);
            }
        } catch (SQLException e) {
            throw new DataProcessingException("Can't get manufacturer from DB. ID: " + id, e);
        }
        return Optional.of(manufacturer);
    }

```